BuildingBlocks/src/BuildingBlocks.Infrastructure/Persistence/BlobStorage/GoogleCloudStorage/GoogleCloudStorageHealthCheck.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Text;
+using Backbone.Tooling.Extensions;
+using Google.Cloud.Storage.V1;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Backbone.BuildingBlocks.Infrastructure.Persistence.BlobStorage.GoogleCloudStorage;
+
+public class GoogleCloudStorageHealthCheck : IHealthCheck
+{
+    public GoogleCloudStorageHealthCheck(StorageClient storage, GoogleCloudStorageOptions options)
+    {
+        _storage = storage;
+        _options = options;
+    }
+
+    private readonly StorageClient _storage;
+    private readonly GoogleCloudStorageOptions _options;
+
+    private static bool? _isHealthy;
+    private const string FILE_TEXT = "healthcheck";
+
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        if (!_isHealthy.HasValue)
+        {
+            try
+            {
+                //Upload random Blob with predefined text
+                var filename = Guid.NewGuid().ToString();
+                await _storage.UploadObjectAsync(_options.BucketName, filename, "text/plain", new MemoryStream(FILE_TEXT.GetBytes()), null, cancellationToken);
+
+                //Download the Blob again and check the text
+                var downloadStream = new MemoryStream();
+                await _storage.DownloadObjectAsync(_options.BucketName, filename, downloadStream, null, cancellationToken);
+                var downloadedString = Encoding.UTF8.GetString(downloadStream.ToArray());
+                if (downloadedString != FILE_TEXT) _isHealthy = false;
```

Comment:
Setting `isHealthy` to false here doesn't do anything. It's overwritten in line 41 or 45 (depending on whether an exception will be thrown by the deletion or not)

I'd suggest to refactor the whole method anyway:

```c#
        if (!_isHealthy.HasValue)
        {
            var isUploadPossible = await IsUploadPossible();
            var isDownloadPossible = await IsDownloadPossible();
            var isDeletionPossible = await IsDeletionPossible();

            _isHealthy = isUploadPossible && isDownloadPossible && isDeletionPossible;
        }

        return _isHealthy.Value ? HealthCheckResult.Healthy() : HealthCheckResult.Unhealthy();
```

This has 2 advantages:
1. It is more readable, because the methods are smaller.
2. You cannot make a mistake like the one I mentioned above, because each method is responsible for its own health check result.